cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartDownscaleStopInstancesHandler.java

```diff
@@ -0,0 +1,90 @@
+import com.sequenceiq.cloudbreak.cloud.model.CloudVmInstanceStatus;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleStopInstancesRequest;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleStopInstancesResult;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartDownscaleStopInstancesHandler implements CloudPlatformEventHandler<StopStartDownscaleStopInstancesRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopStartDownscaleStopInstancesHandler.class);
+
+    private static final Long STOP_POLL_TIMEBOUND_MS = 300_000L;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartDownscaleStopInstancesRequest> type() {
+        return StopStartDownscaleStopInstancesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartDownscaleStopInstancesRequest> event) {
+        StopStartDownscaleStopInstancesRequest request = event.getData();
+        LOGGER.info("StopStartDownscaleStopInstancesHandler: {}", event.getData().getResourceId());
+
+        CloudContext cloudContext = request.getCloudContext();
+        try {
+            CloudConnector<?> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            List<CloudInstance> cloudInstancesToStop = request.getCloudInstancesToStop();
+            List<CloudVmInstanceStatus> cloudVmInstanceStatusList = Collections.emptyList();
+            if (cloudInstancesToStop.size() > 0) {
+                LOGGER.info("Attempting to stop instances with a timebound of {}ms. count={}, instances=[{}]",
+                        STOP_POLL_TIMEBOUND_MS, cloudInstancesToStop.size(), cloudInstancesToStop.stream().map(CloudInstance::getInstanceId)
+                                .collect(Collectors.toList()));
+                // TODO CB-14929: Error handlnig. Currently - an error from the following command (e.g. NPE)
+                //  results in no WARNINGS on the console. The stop is just ignored, and the cluster goes into a
+                //  RUNNING state, with the nodes in a DECOMMISSIONED state. The error handling needs to make sure
+                //  that the system goes into a failed state, and there is at least some logging on the console
+                //  to indicate what happened.
+                // TODO CB-14929: Also, after re-provisioning nodes via CM, the CB UI does not seem to be refreshing the node
+                //  status to RUNNING, and instead the nodes stay in a STUCK state. Is the CM sync task not taking care of this?
+                // TODO CB-15132: What happens if the cloud provider does not know about an instance for which a STOP was requested. How does the API
```

Comment:
you can check, but I think it will throw error. Anyway it should be in deleted state because cloud sync should realize this node is deleted.

Reply:
This (anything CB-15132) is part of extensive error handling to happen in the next phase.
Even if the cloud-sync handler were to run just before this operation ... there's always the possibility of timing issues with the resource not being available. Hence the TODOs to track this down in more detail. As an example - if this were a SPOT instance, there's a higher likelihood of the instance disappearing in the middle of an operation.